pyccel/codegen/printing/ccode.py

```diff
@@ -2437,11 +2437,11 @@ def _print_Return(self, expr):
                 # make sure that stmt contains one assign node.
                 last_assign = last_assign[-1]
                 variables = last_assign.rhs.get_attribute_nodes(Variable)
-                unneeded_var = not any(b in vars_in_deallocate_nodes or b.is_ndarray for b in variables) and \
+                unneeded_var = not any(b in vars_in_deallocate_nodes or b.rank for b in variables) and \
```

Comment:
Could you please explain this change? Does this allow handling sets in addition to arrays?

Reply:
Yes exactly. This enables handling sets, dits, and lists as well as arrays. In other words it now handles any object which needs deallocating